### PR TITLE
LPS-202682 Fix performance and inconsistency problems with nonce handling

### PR DESCRIPTION
**Pull request by izaera**

### CODE REVIEW

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:test:stable

**Comment by izaera** [comment]
ci:test:relevant

File: modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/ContentSecurityPolicyNonceManager.java [hunk 1]
```diff
@@ -27,57 +27,64 @@
...
 		if (!contentSecurityPolicyConfiguration.enabled()) {
-			return StringPool.BLANK;
+			nonce = StringPool.BLANK;
 		}
+		else if (PropsValues.JAVASCRIPT_SINGLE_PAGE_APPLICATION_ENABLED) {
+			HttpSession httpSession = httpServletRequest.getSession();
 
-		httpServletRequest = _portal.getOriginalServletRequest(
-			httpServletRequest);
+			nonce = (String)httpSession.getAttribute(_NONCE);
```
Comment (dsanz) [inline modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/ContentSecurityPolicyNonceManager.java:65]: Wouldn't we like to include getting the nonce logic inside the `synchronized` block? This way we won't allow 2 threads to get a null value before trying to set it

Comment (dsanz) [reply modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/ContentSecurityPolicyNonceManager.java:65]: just realized you did it inside the syncrhonized. So pls disregard my previoius comment

Comment (izaera) [reply modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/ContentSecurityPolicyNonceManager.java:65]: :+1:

**Comment by izaera** [comment]
Since this is OK, I think we can forward it without performance QA looking at it.

In the worst case scenario (performance assesment is still wrong), we would need to make more tweaks, but the content in this PR would still be valid, since it is fixing broken things too.

**Comment by izaera** [comment]
ci:force:forward

**Comment by izaera** [comment]
ci:forward:force

**Pull request closed without merging**